packages/compass-components/src/components/databases-navigation-tree.spec.tsx

```diff
@@ -0,0 +1,369 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
```

Comment:
Having this component here implies a broader architectural decision, would be great if compass/components could only contain domain-free components and raw building blocks, that's how we are approaching this so far at least.

Breaking this pattern has a good potential to create confusion and we won't have anymore a clear cut as for when to put something in components and when not. For example by the same token the connection form could have ended up here. 

Is there a reason (other than ts) why this should not be in the sidebar?

Reply:
I was assuming the only goal with this package is not to have any direct connection to compass internals (which some hadron components have for example), thanks for clarifying that! The main reason was indeed that converting the whole sidebar plugin codebase to the new ts, code formatting, linting, and testing setup would be too much work upfront and as this side navigation is just a react component, putting it here made sense to me. I'm happy to move all this to the separate new monorepo package or work on converting the plugin to our new tooling, whatever makes more sense in your opinion.

I'm also totally okay with following the "only contain domain-free components and raw building blocks", but not really sure though why we are doing it. From my POV if keeping plugins (and tbh most of the other packages in the monorepo) in separate packages was more or less a decision based on how they were published before, consumed by Compass, and the fact that they have slightly different build/test setups, for the new components I don't really see a reason to create multiple packages like that, so I wonder what might be the motivation that I'm missing.

Reply:
Yeah I totally get your point, if is not a big deal for you I'd stick to the pattern where compass-components just domain-free for now. Practically speaking honestly is fine to do this in a separate PR.

I don't mean to be too annoying here, and we can change this rule, but this is my rationale: despite of how good this convention is, and how fit is the current "plugin-based" design with the monorepo, for me we should stick with it until we consciously decide to change, and if and when we do, we should change things everywhere and consistently.

To me is really important that we don't take this decision lightly, and we don't make it happen by accident or as a byproduct of something else (ie `ts`) cause essentially it would create a situation where 2 different "frontends" would exist, one in the plugins and one in `compass-components`, and it since it won't be fun to undo it ..  if that's the direction we want to take we should also make a commitment and a plan to move to only one, at least not leaving it to the case.

> I'm totally okay with following the "only contain domain-free components and raw building blocks", but not really sure though why we are doing it. From my POV if keeping plugins (and tbh most of the other packages in the monorepo) in separate packages was more or less a decision based on how they were published before, consumed by Compass, and the fact that they have slightly different build/test setups, for the new components I don't really see a reason to create multiple packages like that, so I wonder what might be the motivation that I'm missing.

I think here there are 2 design decisions involved though:
- The first decision is for `compass-components`: this one is basically a central place for reusable UI components, is a way to change in one place how the application looks, kinda a re-export from LG + the components we miss from LG, but that's the level and purpose, just a way to factor together duplicate styles.

- The second one is about the plugins: i don't think the separation they create is so terrible, but i get that we don't need to have them in 100 packages. Also we have many that do not make so much sense. The general fact that we have things with significantly different state / doing different actions and with an independent set of functionalities separated makes sense to me, but yeah those could be just different folders in a package rather then different packages and i'd be ok with that.

Personally i think those should be 2 different decisions, for example if we remove the packages then we could make everything just a folder inside the same package, but `compass-components` could still be a subfolder, we would import `<Button>` and `colors.gray1` from there and `sidebar` could be another one.

Reply:
> if is not a big deal for you I'd stick to the pattern where compass-components just domain-free for now

Yeah, absolutely, will move it out, thanks for sharing your thoughts on this.

Reply:
Done in 22c421e, but tbh I'm not sure if I should've moved more or less stuff, in theory useTree hooks are pretty generic, but then also would we really have another navigation tree view in the app? Idk, anyway, tell me what you think and if this matches what you were expecting